src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,83 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  postId: number;
+  addCommentOnServer: (newComment: Comment) => Promise<void>;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  postId,
+  addCommentOnServer,
+}) => {
+  const [nameErr, setNameErr] = useState(false);
+  const [emailErr, setEmailErr] = useState(false);
+  const [textErr, setTextErr] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [text, setText] = useState('');
+
+  const validation = () => {
+    let isPassValid = true;
+
+    if (!name.trim()) {
+      setNameErr(true);
+      isPassValid = false;
+    }
+
+    if (!text.trim()) {
+      setTextErr(true);
+      isPassValid = false;
+    }
+
+    if (!email.trim()) {
+      setEmailErr(true);
+      isPassValid = false;
+    }
+
+    if (!isPassValid) {
+      return false;
+    }
+
+    return true;
+  };
+
+  const addComment = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (!validation()) {
+      return;
+    }
+
+    setIsLoading(true);
+    addCommentOnServer({
+      id: 0,
+      name,
+      postId,
+      body: text,
+      email,
+    })
+      .then(() => {
+        setText('');
+      })
+      .finally(() => {
```

Comment:
any cath error logic?

Reply:
Yas,  cath error logic relyse on PostDetails. Hear this constraction inly for clean form and set/remove loader. Notification user in PostDetails